File: rvItemControllerSource/src/main/java/com/clumob/listitem/controller/source/EmptyItemSource.java
package com.clumob.listitem.controller.source;

/**
 * Created by prashant.rathore on 17/12/18.
 */

public class EmptyItemSource extends ItemControllerSource<ItemController> {


    @Override
    public void onAttached() {

    }

    @Override
    public void onItemAttached(int position) {

    }

    @Override
    public boolean hasStableIds() {
        return false;
    }

    @Override
    public int getItemPosition(ItemController item) {
        return 0;
    }

    @Override
    public ItemController getItemForPosition(int position) {
        return null;
    }

    @Override
    public void onDetached() {

    }

    @Override
    protected int computeItemCount() {
        return 0;
    }
}

File: rvItemControllerSource/src/main/java/com/clumob/listitem/controller/source/EmptyItemSource.kt
package com.clumob.listitem.controller.source

/**
 * Created by prashant.rathore on 17/12/18.
 */
class EmptyItemSource : ItemControllerSource<ItemController>() {
    override fun onAttachToView() {}
    override fun onItemAttached(position: Int) {}
    override fun hasStableIds(): Boolean {
        return false
    }

    override fun getItemPosition(item: ItemController): Int {
        return 0
    }

    override fun getItemForPosition(position: Int): ItemController {
        throw Exception("Its an EmptySource. Should not come here $position")
    }

    override fun onDetachFromView() {}
    override fun computeItemCount(): Int {
        return 0
    }
}